enterprise/backend/src/metabase_enterprise/semantic_search/db/connection.clj

```
@@ -0,0 +1,18 @@
+(ns metabase-enterprise.semantic-search.db.connection
+  (:require
+   [metabase-enterprise.semantic-search.db.datasource :as semantic.db.datasource]
+   [metabase-enterprise.semantic-search.db.locking :as semantic.db.locking]
+   [next.jdbc :as jdbc]))
+
+(defn do-with-migrate-tx
```

In my view this macro does not add anything and makes the code less precise and direct.

- The important operation is the acquisition of the migration lock. Callers should call that. Then you do not need this namespace at all, if the `#_` are deleted?
- Calling that directly rather than using this macro
   * leaves a plain specification of the transaction in the code using the expected `next.jdbc` calls
   * no footgun prone conn defaulting (what if the call is wrong? We would throw away our existing transaction policy and grab another thread from the global connection pool - silently succeed but cause some race / serialization issue, hit the wrong database in a test, or not get the right connection setup or xact variables)
- As it is an xact lock there is no special lifetime to manage (transaction anyway), so not sure why we need another scope

---

I'd rather remove the namespace when there's liquibase follow-up. Not yet clear on specifics of that, but this leaves an option to eg. set the timeout on the connection. If it turns out we don't need it then, the whole namespace can go.